pkg/controllers/elasticsearch/nodepool/controller.go

```diff
@@ -172,40 +172,34 @@ func (e *statefulControl) existingStatefulSet(c *v1alpha1.ElasticsearchCluster,
 	return sets[0], nil
 }
 
-func (e *statefulControl) syncPilotResources(c *v1alpha1.ElasticsearchCluster, np *v1alpha1.ElasticsearchClusterNodePool, ss *appsv1beta1.StatefulSet) error {
-	// TODO: use labels to limit which pods we list to save memory
-	allPods, err := e.podLister.Pods(c.Namespace).List(labels.Everything())
+func (e *statefulControl) syncPilotResources(c *v1alpha1.ElasticsearchCluster) error {
+	selector, err := util.SelectorForCluster(c)
+	if err != nil {
+		return err
+	}
+
+	allPods, err := e.podLister.Pods(c.Namespace).List(selector)
 	if err != nil {
 		return err
 	}
+
 	for _, pod := range allPods {
 		isMember, err := util.PodControlledByCluster(c, pod, e.statefulSetLister)
 
 		if err != nil {
 			return fmt.Errorf("error checking if pod is controller by elasticsearch cluster: %s", err.Error())
 		}
 
-		if isMember {
```

Comment:
Isn't it important to *also* check that the pod is owned by the esCluster resource?
I thought this check was to ensure against deleting pods that, *somehow*, had been created independently, with the matching cluster label.
And if not, we've got code that looks up the owner of the owning statefulset; can that now be removed.

Reply:
Yep - this shouldn't have been removed. I've added that check back in.